### ISSUE TITLE
TypeError, possibly due to corrupt stream

### ISSUE THREAD

**Issue by atombender** [regular/unbzip2-stream#3]
There's a possibility that my data is corrupt. Regardless, the stream shouldn't throw a random error like this.

```
TypeError: Cannot read property '0' of undefined
   at f (/tmp/app/node_modules/unbzip2-stream/lib/bit_iterator.js:24:34)
   at Object.bzip2.decompress (/tmp/app/node_modules/unbzip2-stream/lib/bzip2.js:272:13)
   at decompressBlock (/tmp/app/node_modules/unbzip2-stream/index.js:29:28)
   at decompressAndQueue (/tmp/app/node_modules/unbzip2-stream/index.js:46:20)
   at Stream.end (/tmp/app/node_modules/unbzip2-stream/index.js:86:23)
   at _end (/tmp/app/node_modules/unbzip2-stream/node_modules/through/index.js:65:9)
   at Stream.stream.end (/tmp/app/node_modules/unbzip2-stream/node_modules/through/index.js:74:5)
   at PassThrough.onend (_stream_readable.js:501:10)
   at PassThrough.g (events.js:180:16)
   at PassThrough.emit (events.js:117:20)
   at _stream_readable.js:943:16
   at process._tickDomainCallback (node.js:463:13)
```

**Comment by regular** [comment]
Can you upload the offending file somewhere? Will be hard to reproduce otherwise.

**Comment by atombender** [comment]
Unfortunately, it happens seemingly randomly among a huge load of files archived on S3. I've since moved to a native bzip2 NPM module, since yours proved to be about 10x slower than native.

**Comment by tedbrandston** [comment]
I'm seeing the same thing for basically any truncated bz2 file I make. There will be slight differences in the location in bzip2.js (line 272, line 277, etc.) depending on exactly where the file is cut off.

**Comment by tedbrandston** [comment]
looks like it's because of the filtering of `Bzip2Error`s from other errors [here](https://github.com/regular/unbzip2-stream/blob/master/index.js#L56). Looks like it's been broken since the [initial commit](https://github.com/regular/unbzip2-stream/commit/9d308eae609e6272e4b1f107706afef3ff1763c0#diff-168726dbe96b3ce427e7fedce31bb0bcR68)?

Not sure if there's a good reason to do it that way, so I'll put in a pull request.

@atombender what did you end up switching to?

**Comment by atombender** [comment]
@tedbrandston: Switched to [this](https://github.com/atheken/gzbz). No complaints so far.

**Comment by regular** [comment]
Fixed in 1.0.10
